File: java/src/com/zhy/DesignPattern/AbstractFactory/BenzFactory.java
package com.zhy.DesignPattern.AbstractFactory;

public class BenzFactory implements ICarFactory {

	@Override
	public IBody CreateBody() {
		return new BenzBody();
	}
	
	@Override
	public IEngine CreateEngine() {
		return new BenzEngine();
	}
	
	@Override
	public IWheel CreateWheel() {
		return new BenzWheel();
	}
}

File: csharp/DesignPattern.AbstractFactory/Factory/BenzFactory.cs

using DesignPattern.AbstractFactory.ProductBody;
using DesignPattern.AbstractFactory.ProductEngine;
using DesignPattern.AbstractFactory.ProductWheel;
namespace DesignPattern.AbstractFactory.Factory
{
    /// <summary>
    /// 具体的奔驰工厂类
    /// </summary>
    public class BenzFactory : ICarFactory
    {
        public IEngine CreateEngine()
        {
            return new BenzEngine();
        }

        public IWheel CreateWheel()
        {
            return new BenzWheel();
        }

        public IBody CreateBody()
        {
            return new BenzBody();
        }
    }
}
